### ISSUE TITLE
Solution: how to load templates dynamically

### ISSUE THREAD

**Issue by swift1** [vuejs/Discussion#313]
This took me several hours to figure out, so I thought I'd share the solution with you, since this is quite a common question when peoples' applications get large.

First, make sure you include the http resource loader:
`````html
<script src="https://cdnjs.cloudflare.com/ajax/libs/vue-resource/0.1.10/vue-resource.min.js"></script>
````

Then, make a html file and put all your template markup there. For example async-component.html.

***DO NOT*** wrap the html inside a script tag like this:
````html
<script id="someAsyncTemplate" type="text/x-template"></script>
````

In the javascript, (which could be yet another external file with ````<script src="some-file.js"></script>````) write the component exactly as you normally would, but wrap the whole thing inside the following code:

````javascript
Vue.component('async-component', function (resolve, reject) {
	vue.$http.get('async-component.html', function(data, status, request){
		var parser = new DOMParser();
		var doc = parser.parseFromString(data, "text/html");
		resolve({
			template: doc
		});
	});
});
````

so basically, the whole component code must be inside the resolve-callback, like this:
````resolve({ <data, props, mixins, methods, computeds, directives...> })````.

***Note:***
*a) This code assumes that you have created a reference to Vue called ````vue````, like this:*
````
var vue = new Vue({
    ...
});
````
*b) In order to prevent weird side-effects with international characters not being rendered correctly when rendering the html from the external file, make sure the html file is saved as "UTF-8 without BOM".*

**Comment by PavelPolyakov** [comment]
Good to know, thanks

**Comment by bart** [comment]
Hey, I am searching for a solution loading the template for a component dynamically. But when I try your suggestion, I'm getting an error: `Uncaught TypeError: Cannot read property '$http' of undefined`

My code looks something like this:

```
Vue.component('async', function (resolve, reject) {
    vue.$http.get('async-component.html', function(data, status, request){
        var parser = new DOMParser();
        var doc = parser.parseFromString(data, "text/html");
        resolve({
            template: doc,
            ready: function() {
                console.log('I am ready!');
            }
        });
    });
});

var vue = new Vue({
    el: 'body'
});
```

Do you have any idea?

**Comment by swift1** [comment]
@bart make sure you include the http resource loader:

``` html
<script src="https://cdnjs.cloudflare.com/ajax/libs/vue-resource/0.1.10/vue-resource.min.js"></script>
```

You also need the vue variable:

```
var vue = new Vue({
    el: 'body'
});
```
